File: SpiderEngine/Model/SpiderParserInfo.cs
namespace SpiderEngine.Model;

public class SpiderParserInfo
{
    public SpiderParserType Type { get; set; } = default!;
    public string Patten { get; set; } = default!;
    public IEnumerable<SpiderKeyPathPair> Value { get; set; } = default!;
}

public enum SpiderParserType
{
    Regex,
    Json,
}

File: android/src/main/kotlin/com/itoken/team/wutspider/model/SpiderParserInfo.kt
package com.itoken.team.wutspider.model

import kotlinx.serialization.ExperimentalSerializationApi
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonNames

@Serializable
data class SpiderParserInfo @OptIn(ExperimentalSerializationApi::class) constructor(
    val type: SpiderParserType,
    @JsonNames("patten") val pattern: String? = null,
    val value: List<SpiderKeyPathPair>? = null
)

@Serializable
enum class SpiderParserType {
    @SerialName("regex") REGEX,
    @SerialName("json") JSON
}